sklearn/metrics/_scorer.py

```diff
@@ -371,6 +376,45 @@ def set_score_request(self, **kwargs):
             self._metadata_request.score.add_request(param=param, alias=alias)
         return self
 
+    def _get_metadata_request(self):
+        """Get requested data properties.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        Returns
+        -------
+        request : MetadataRequest
+            A :class:`~sklearn.utils.metadata_routing.MetadataRequest` instance.
+        """
+        if hasattr(self, "_metadata_request"):
+            requests = get_routing_for_object(self._metadata_request)
+        else:
+            requests = MetadataRequest(owner=self)
+            setattr(
+                requests,
+                "score",
+                MethodMetadataRequest(
+                    owner=self,
+                    method="score",
+                    requests=self._get_class_level_metadata_request_values(
```

Comment:
I might need some context to understand these two cases. When is the first condition applicable and when is the second one?

Reply:
As soon as a user calls `set_score_request`, the values are stored in the `_metadata_request` attribute, otherwise the attribute doesn't exist and we always return the default request values. This is exactly the same as what happens for all consumer estimators.